ros2_generate_interface_docs/ros2_generate_interface_docs/utils.py

```diff
@@ -0,0 +1,153 @@
+# Copyright 2020 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from io import StringIO
+import os
+import sys
+
+import em
+
+from rosidl_parser.definition import AbstractNestedType, BASIC_TYPES, NamespacedType
+
+_TEMPLATES_DIR = 'templates'
+IGNORED_KEYS = ['__slots__', '__doc__', '_fields_and_field_types', 'SLOT_TYPES', '__init__',
+                '__repr__', '__eq__', '__hash__', 'get_fields_and_field_types', '__module__']
+
+
+def generate_raw_text(raw_text):
+    raw_test_str = ''
+    for line in raw_text.split(os.linesep):
+        line = line.replace(' ', '&nbsp;')
+        parts = line.split('#')
+        if len(parts) > 1:
+            raw_test_str = raw_test_str + parts[0] \
+                + '<font color="blue">#%s</font><br/>' % ('#'.join(parts[1:]))
+        else:
+            raw_test_str = raw_test_str + '%s<br/>' % parts[0]
+    return raw_test_str
+
+
+def resource_name(resource):
+    """Return the resource name."""
+    if '/' not in resource:
+        return '', '', resource
+    values = resource.split('/')
+    if len(values) != 3:
+        raise ValueError(...)
```

Comment:
This is missing an error message

Reply:
2fc43c7162392cf15199cee6a2d515b40a798aa8